spec/features/rag_wise/forum_post_spec.rb

```diff
@@ -256,6 +277,19 @@
         topic_post = topic.posts.first
         expect(topic_post).to_not have_selector('div.MuiChip-root', text: 'Generate reply')
       end
+
+      scenario 'I can trigger automatic response by posting new post' do
+        visit course_forum_topic_path(course, forum, topic)
+        topic_post = topic.posts.first
+
+        find("button.post-reply-#{topic_post.id}").click
+
+        # Reply a post with the default title.
+        fill_in_react_ck "textarea[name=postReplyText_#{topic_post.id}]", 'test'
+        expect_any_instance_of(Course::Forum::PostsController).to receive(:auto_answer_action).once
+        find('.reply-button').click
+        wait_for_page
```

Comment:
Why click a button and `wait_for_page` at the end of the test?

We aren't expecting anything after.

Same for L211-L212

Reply:
Removing wait_for_page makes the test cases flaky as it might not receive the signal for auto_answer yet